### PR TITLE
SG-13266: port to Python 3

### PR DESCRIPTION
**Pull request by jfboismenu**
Ported the framework to Python 3. Note that the updates to the `build_resources.sh` script and the resources are on [another branch](https://github.com/shotgunsoftware/tk-framework-qtwidgets/compare/SG-13266_clean_review?expand=1), otherwise there would be another 100+ files in the PR.

I added a test that loads the framework and instantiates all the widgets one by one. I actually found some bugs with it, so it was useful.

Then I started working on the multi demo app, which helped me find more issues, which I've fixed in this branch.

### CODE REVIEW

File: python/activity_stream/data_manager.py [hunk 1]
```diff
@@ -722,37 +726,39 @@ def __db_insert_activity_updates(
                         INSERT OR REPLACE INTO activity(activity_id, payload, created_at)
                         SELECT ?, ?, datetime('now')
                     """
+                    params = (activity_id, blob)
```
Comment (jfboismenu) [inline python/activity_stream/data_manager.py]: This was actually a bug in the current implementation! We passed too many bindings in.

File: python/models/hierarchical_filtering_proxy_model.py [hunk 2]
```diff
@@ -141,7 +142,7 @@ def minimize(self):
                 return
 
             self._cache = dict(
-                [(k, v) for k, v in self._cache.iteritems() if v[0].isValid()]
+                [(k, v) for k, v in six.iteritems(self._cache) if v[0].isValid()]
```
Comment (jfboismenu) [inline python/models/hierarchical_filtering_proxy_model.py]: Filtering model need to be super-fast, so I'm keeping the use of iterators here.

File: tests/fixtures/config/env/test.yml [hunk 3]
```diff
@@ -14,7 +14,7 @@ engines:
       type: path
       path: $SHOTGUN_TEST_ENGINE
 frameworks:
-  tk-framework-qtwidgets:
+  tk-framework-qtwidgets_v2.x.x:
```
Comment (jfboismenu) [inline tests/fixtures/config/env/test.yml]: This is from the test that imports everything I mentionned in the PR description.

File: tests/test_framework.py [hunk 4]
```diff
@@ -8,22 +8,147 @@
...
+                        params["bg_task_manager"] = bg_task_manager
+                    elif arg == "fields_manager":
+                        params["fields_manager"] = field_manager.ShotgunFieldManager(
+                            parent=parent, bg_task_manager=bg_task_manager
+                        )
+                    elif arg == "sg_entity_type":
+                        params["sg_entity_type"] = "Asset"
+
+                # Add the widget and show it, don't worry about the rest.
+                # It. Just. Works.
```
Comment (willis102) [inline tests/test_framework.py]: 😂 ❤️

**Pull request merged**